src/funjson.c

```diff
@@ -64,6 +65,36 @@ json_escape_string(char *input)
   return buff;
 }
 
+/** Unescape a string for use as a JSON string. Returns a calloced buffer under 'json.string.partialbuf'. **/
+char * json_unescape_latin1string(char *latin1, int len)
+{
+  char *c;
+  char *partialbuf = mush_calloc(len*6,sizeof(char),"json.string.partialbuf");
+  char *bpp = partialbuf;
+  int pblen = 0;
+  do_log(LT_WIZ,GOD,GOD,"%d",UNSAFE_UNESCAPE);
+  for (c = latin1; *c; c += 1) {
+    if(*c == TAG_START && UNSAFE_UNESCAPE) {
+      safe_strl(MARKUP_START, strlen(MARKUP_START), partialbuf, &bpp);
+      pblen += strlen(MARKUP_START);
+    }
+    else if(*c == TAG_END && UNSAFE_UNESCAPE) {
+      safe_strl(MARKUP_END, strlen(MARKUP_END), partialbuf, &bpp);
+      pblen += strlen(MARKUP_END);
+    }
```

Comment:
You probably want to ignore everything between TAG_START and TAG_END when unsafe_escape is on, not just those specific characters. You should also be checking for ESC_CHAR.

Reply:
Fair, with the changes, I can also grab ansi codes.